randovania/games/game.py

```diff
@@ -8,6 +8,11 @@
 
 import randovania
 from randovania.bitpacking.bitpacking import BitPackEnum
+from randovania.generator.filler.weights import (
```

Comment:
Slightly concerned with this file importing something from the `randovania.generator` module, outside of the TYPE_CHECKING block. This might cause some import loops in weird scenarios?

Tbh looking at it we really should split this file huh? No reason for `GameData` to come from the same file as `RandovaniaGame`.

Reply:
I've been considering not having default values, but I thought that for the sake of having it easier for people to configure this'd be nicer.

But if every new game just comes from BDG, I guess that argument doesnt really hold. So there would just be the move of generator default weigths from game.py to games/[game_name]/game.py which I don't think would change much

---

I did initially have the weights in retcon.py but moved them out of there cause they caused circular dependencies. Weights being in their own file is probably fine? Cause that constants file cant really import anything else.

Reply:
Oh no I like the default values. Let's use them so we can adjust the generator in a global way without having to adjust every single game.

Importing the separate file still imports every `__init__.py` in the way, which would cause issues if these had their own code.

I like to think of the different folders in the module root as it's own modules and to keep an uni-directional dependency loop. So if module A imports module B, then module B is not allowed to import anything from module A ever.

At least, not outside of TYPE_CHECKING blocks.

Reply:
I see. I don't think there's a nice way around this though. I was thinking of maybe having weights as its own class instead of just random constants, and then GameData can just reference this for type checking, but since it would still need to use the default values/class, it would precisely need to import it *not* as type checking.